### DISCUSSION TITLE
Servicemeldung Intervall

### DISCUSSION THREAD

**Discussion by ps915** [SukramJ/aiohomematic#1848] [Q&A]
### The problem

Die Service Meldungen in HA wird der Servicemeldungen Counter erst dann aktualisiert, wenn ich mich in RaspberryMatic einlogge. Heute habe ich bemerkt, dass die Entität Servicemeldung 11 Stunden lang den selben Wert hatte, in RaspberryMatic war die Fehlermeldung aber schon lange weg (selbst quittiert)

Ist das normal, dass sich die Servicemeldung Entität so verhält?

### What version of HomematicIP (local) has the issue?

1.70.0

**Comment by SukramJ** [comment]
Die "Anzahl der Servicemeldungen" ist eine Systemvariable, die im default, zusammen mit den anderen Systemvariablen, alle 30 Sekunden abgerufen wird.
Die HA Anzeige wird nur aktualisiert, wenn der Wert sich ändert, aber ich kann nicht sagen wann die Systemvariable Servicemeldungen in der CCU aktualisiert wird

**Comment by ps915** [comment]
Okay, bei mir wird dieser Wert allerdings nicht korrekt aktualisiert. Erst zuletzt war die letzte Aktualisierung 11 Stunden her und die CCU3 hatte schon längst den Fehler quittiert. 

Scheint also ein Bug zu sein?

**Reply by SukramJ** [reply]
> Scheint also ein Bug zu sein?

Du darfst nicht das CCU UI Element Servicemeldungen mit der Systemvariablen Servicemeldungen verwechseln.
Hier wird nur die Systemvariable angezeigt.

> aber ich kann nicht sagen wann die Systemvariable Servicemeldungen in der CCU aktualisiert wird

**Comment by ps915** [comment]
Ich rede von diesen beiden "Sensoren"

![image](https://github.com/user-attachments/assets/eeda33a9-fbc5-450e-9444-c056391abb4b)

Die zeigen mir doch die aktuelle Anzahl an Meldungen an, oder?

**Reply by SukramJ** [reply]
Ja, aber ich wiederhole noch mal:

Du darfst nicht das CCU UI Element Servicemeldungen mit der Systemvariablen Servicemeldungen verwechseln.
Hier wird nur die Systemvariable angezeigt.

**Comment by ps915** [comment]
Okay, das wusste ich nicht. Ist auch etwas verwirrend. 

Das heißt:
- wenn gerade aktuell eine oder mehrere Systemmeldungen im RaspberryMatic UI präsent sind, sehe ich das nicht sofort in HA?
- bzw nach dem quittieren in der Raspberry Matic UI sehe ich diesen Zustand auch nicht in HA?

![image](https://github.com/user-attachments/assets/b6d2ae8f-c95f-425d-b05a-46031c84475d)

**Reply by SukramJ** [reply]
Ich kenne die Systematik in der CCU nicht.

Du denkst, das was in HA als Sensor zu sehen ist entspricht genau der CCU UI anzeige, und das ist falsch.
Ich kann dir nicht sagen, wie die CCU die Systemvariable befüllt!

**Comment by ps915** [comment]
Hier ein aktuelles Beispiel. 
Home Assistant zeigt seit einer Stunde eine 1 an. 

Im UI von Raspberry Matic ist keine Servicemeldung zu sehen. Ebenfalls ist nichts in den Systemvariablen hinterlegt. 

Was soll mir die Entität in HA nun sagen? 

![Screenshot_20241118-182751](https://github.com/user-attachments/assets/a3e54651-e4b6-48cc-8c74-9d3769f83562)
![Screenshot_20241118-182808](https://github.com/user-attachments/assets/c5b05952-ce5d-4abe-9fde-69c491822f44)

**Reply by SukramJ** [reply]
Falls es noch nicht richtig rübergekommen ist:
1. Es ist mir egal was die UI anzeigt
2. Ich weiss nicht, wie die Systemvariable beschrieben wird.
3. Es wird nur der Wert der Systemvariable `Servicemeldung` angezeigt.
4. Es ist nur eine Systemvariable, die von deiner CCU bereitgestellt wird, die den Namen `Servicemeldung` hat.
5. Für alle weiteren Fragen zu der Systemvariable mit dem Namen `Servicemeldung` bist Du hier falsch.

**Comment by Baxxy13** [comment]
Also die SysVar heißt auf der Zentrale "Servicemeldungen" und enthält als Wert die Anzahl der anstehenden Servicemeldungen, so wie man die auch im Kopf der WebUI bei Servicemeldungen (x) sieht.
Der Zähler wird instant aktualisiert, d.h. ist die letzte Servicemeldung beseitigt geht der Wert der "Servicemeldungen" auf "0".

So auch auf meinem HA-Testsystem zu sehen, natürlich teilweise mit Verzögerung die maximal den Wert des eingestellten "SysVar Scan Intervall's" beträgt. 
(default 30s).

Soweit hier also alles i.O.

Was mich jetzt ein bisschen an den Screenshots wundert...
Das Teil heißt da "Servicemeldung" und hat kein Standard-Symbol. Könnte natürlich umbenannt und das Symbol geändert worden sein.

Bei mir heißt die "Sv_Servicemeldungen" und hat das default (Augen) Symbol.

Die Frage wäre also ob da wirklich die "Sv_Servicemeldungen" dahintersteckt oder es doch was anderes ist.

**Reply by ps915** [reply]
> Also die SysVar heißt auf der Zentrale "Servicemeldungen" und enthält als Wert die Anzahl der anstehenden Servicemeldungen, so wie man die auch im Kopf der WebUI bei Servicemeldungen (x) sieht.
> Der Zähler wird instant aktualisiert, d.h. ist die letzte Servicemeldung beseitigt geht der Wert der "Servicemeldungen" auf "0".

So auch auf meinem HA-Testsystem zu sehen, natürlich teilweise mit Verzögerung die maximal den Wert des eingestellten "SysVar Scan Intervall's" beträgt.
(default 30s).

Genau so habe ich es auch verstanden, aber es verhält sich bei mir nicht so.

> Die Frage wäre also ob da wirklich die "Sv_Servicemeldungen" dahintersteckt oder es doch was anderes ist.

Ja genau so ist es. Ich habe lediglich den Namen und den Icon geändert. Die Entitäten-ID ist wie gehabt!

Die Entität zeigt ja auch im Grunde genau das an, was auch im Raspberrymatic UI steht (Anzahl an zu quittierenden Servicemeldungen), nur eben nicht alle 30 Sekunden, sondern "hängt" teileweise mehrere Stunden auf der selben Anzahl an Meldungen fest. 

![image](https://github.com/user-attachments/assets/51f173ac-3f77-4f72-befb-3537cde9a0ee)

**Reply by Baxxy13** [reply]
> "hängt" teileweise mehrere Stunden

Kannst du "teilweise" präzisieren?
Wie ist dein "SysVar Scan Intervall" eingestellt? 30s?

**Reply by ps915** [reply]
Ich benutze die Entität `sensor.ccu3_sv_servicemeldungen` in Home Assistent, damit ich sehe, wenn eine Meldung auf der CCU3 besteht.
In der Regel sind das kurzzeitige Kommunikationsstörungen, die die CCU3 selber quittiert. Das heißt, diese Meldungen sind wahrscheinlich nur ein paar Sekunden oder Minuten vorhanden. 

![image](https://github.com/user-attachments/assets/ce2c4bf6-9094-4f8f-a3d4-2e7a0403b125)


In der Regel, logge ich mich direkt nachdem ich sehe, dass  `sensor.ccu3_sv_servicemeldungen`  größer 0 ist auf der CCU3 Webinterface, um zu schauen, was das für eine Meldung ist. Doch in den seltensten Fällen ist dort noch eine Meldung vorhanden. 

Wie lange genau die Meldungen vorliegen, kann ich nicht sagen. In Home Assistant habe ich aber schon teilweise 11 Stunden lang eine Servicemeldung gehabt, welche natürlich nach dem Login in RaspberryMatic nicht vorhanden war. 

> Wie ist dein "SysVar Scan Intervall" eingestellt? 30s?

Absichtlich habe ich diesen Wert nicht geändert. Wo finde ich diesen Wert, dann prüfe ich das mal!


Ich habe ein Gerät, dessen Batterien leer sind. Ich habe gerade mal den Haken "Servicemeldungen" bei diesem Gerät aktiviert um eine Servicemeldung zu "provozieren".

![image](https://github.com/user-attachments/assets/b8481497-0655-425e-8306-f8d256cc66ff)

In Home Assistent ist die Entität  `sensor.ccu3_sv_servicemeldungen` nahezu gleichzeitig auf 1 gegangen. 

![image](https://github.com/user-attachments/assets/6aac9be2-83dc-4f5e-9971-a9a23f81aff7)

Es scheint also "irgendwie" zu funktionieren. Ich habe das Gefühl, sobald ich eingeloggt in der Raspberry Matic UI bin bzw. im Webinterface etwas tue, werden die Servicemeldungen sehr schnell in HA reflektiert. 

Wenn ich jedoch von der Arbeit nach Hause komme, und ich dann im Verlauf der Entität  `sensor.ccu3_sv_servicemeldungen` die letzte Änderung vor mehreren Stunden sehe, mich dann in Raspberry Matic einlogge ist eine Servicemeldung aktiv.

Nachdem ich die Servicemeldung quittiert habe, schlägt es instant in HA durch.
![image](https://github.com/user-attachments/assets/59a1f1df-c789-4ee7-aa63-35037f30e783)

Es scheint also, dass dieses 30s Intervall irgendwie nicht dauerhaft funktioniert.

**Comment by Baxxy13** [comment]
Aha... du nutzt die Trickserei der RaspberryMatic. 
Du solltest den Haken für Servicemeldungen überall drin lassen und dann funktioniert das auch.

**Reply by ps915** [reply]
Ich habe bei jedem Gerät diesen Haken drin, nur bei dem einen Gerät nicht, da ich das Gerät aktuell nicht nutze und die Batterien leer sind. 

Übrigens, sobald ich die Home Assistant HomematicIP Integration neu starte oder RaspberryMatic neu starte habe ich den aktuellen Wert der Servicemeldungen auch in HA verfügbar. 

Sieht mir danach aus, dass dieses 30 Sekunden Intervall einfach irgendwann stoppt, nach einer gewissen Zeit.
